src/presentation_exchange.rs

```diff
@@ -277,6 +320,71 @@ impl PresentationDefinition {
         self
     }
 
+    /// Validate a presentation submission against the presentation definition.
+    ///
+    /// Checks the underlying presentation submission parsed from the authorization response,
+    /// against the input descriptors of the presentation definition.
+    pub async fn validate_authorization_response(
+        &self,
+        auth_response: &AuthorizationResponse,
+    ) -> Result<()> {
+        match auth_response {
+            AuthorizationResponse::Jwt(_jwt) => {
+                // TODO: Handle JWT Encoded authorization response.
+
+                bail!("Authorization Response Presentation Definition Validation Not Implemented.")
+            }
+            AuthorizationResponse::Unencoded(response) => {
+                let presentation_submission = response.presentation_submission().parsed();
+
+                let jwt = response.vp_token().0.clone();
+
+                let jws = CompactJWSString::from_string(jwt.clone()).context("Invalid JWT.")?;
+                let resolver: VerificationMethodDIDResolver<DIDKey, AnyJwkMethod> =
+                    VerificationMethodDIDResolver::new(DIDKey);
+
+                let params = VerificationParameters::from_resolver(resolver);
+
+                if let Err(e) = jws.verify(params).await {
+                    bail!("JWT Verification Failed: {:?}", e)
+                }
```

Comment:
NOTE: The e2e test is currently failing due to `MissingPublicKey`.

```
unexpected status: Complete(Error { cause: JWT Verification Failed: MissingPublicKey })
```

I'm assuming I am incorrectly signing the verifiable presentation claim? Is this the correct flow?

https://github.com/spruceid/oid4vp-rs/pull/8/files#diff-6d6c7d16f3d6119451ac6b8c431591a112676ff96645d7a6cba9055aa85a7d2aR95-R100

Reply:
Looking at the companion test wallet, it seems I should be using the `Presentation::generate_jwt` from `ssi_vc` to generate the signed JWT.

This brings up another question: should we bump the version of ssi-vc `0.2.*` -> `0.3.*`.

It looks like `Presentation` has become a trait and not a struct type.

For now, I will use the `0.2.1` version and see if I can get the e2e test working correctly.

Reply:
I've removed the JWT encoding and the verification from the e2e test. Using the unencoded flow. Will open a new PR for the JWT processing.